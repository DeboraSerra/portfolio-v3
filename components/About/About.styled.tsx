import styled from "styled-components";
import check from '../../assets/images/about/check-mark-icon.svg';

export const Main = styled.div`
  min-height: 72vh;
  background-color: ${(props) => props.theme.bg.primary};
  padding: 24px 48px;

  color: ${(props) => props.theme.text.secondary};

  .about {
    &__title {
      margin-bottom: 48px;
      text-align: center;
      font-size: ${(props) => props.theme.fontSizes[6]};
      position: relative;

      &::before {
        content: "";
        background-color: ${(props) => props.theme.bg.inset};
        position: absolute;
        bottom: -12px;
        left: 50%;
        transform: translateX(-50%);
        width: 150px;
        height: 1px;
      } // &::before
    } // &__title

    &__left-sect {
      line-height: 1.6;
      text-align: center;
      overflow: hidden;
    } // &__left

    &__p {
      margin-bottom: 12px;

      font-size: ${(props) => props.theme.fontSizes[1]};
      text-align: left;

      &::first-letter {
        font-size: ${(props) => props.theme.fontSizes[3]};
      } // &::first-letter
    } // &__p

    &__img {
      float: right;
      height: 350px;
      object-fit: cover;
      object-position: top left;
      margin: 0 24px;
    } // &__img

    &__skills {
      &-title {
        margin: 48px auto;
        position: relative;

        font-size: ${(props) => props.theme.fontSizes[4]};
        text-align: center;

        &::before {
          content: "";
          background-color: ${(props) => props.theme.bg.inset};
          position: absolute;
          bottom: -12px;
          left: 50%;
          transform: translateX(-50%);
          width: 150px;
          height: 1px;
        } // &::before
      } // &-title

      &-list {
        list-style-position: inside;
        list-style-type: circle;
        display: flex;
        flex-direction: column;
        flex-wrap: wrap;
        gap: 12px;
        max-height: 50vh;

        & li {
          max-width: 200px;

          line-height: 2;
          font-size: ${(props) => props.theme.fontSizes[1]};
        }// & li
      }// &-list
    } // &__skills
  } // .about

  @media only screen and (max-width: 576px) {
    padding: 12px 24px;

    .about {
      &__img {
        float: none;
        margin: 12px 0;
      } // &__img

      &__skills {
        &-list {
          max-height: 80vh;
        }// &-list
      }// &__skills
    } // .about
  } // @media only screen and (max-width : 576px)
`;
