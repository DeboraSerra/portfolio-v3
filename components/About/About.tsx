import { NextPage } from "next";
import Image from "next/image";

import * as S from "./About.styled";
import Fred from "../../assets/images/about/fred.jpg";

interface Props {}

const About: NextPage<Props> = () => {
  const age = () => {
    const birth = new Date("09/16/1992");
    const monthDiff = Date.now() - birth.getTime();
    const ageDt = new Date(monthDiff);
    const year = ageDt.getUTCFullYear();
    const age = Math.abs(year - 1970);
    return age;
  };

  return (
    <S.Main>
      <div className="container">
        <h2 className="about__title">About me</h2>
        <section className="about__left-sect">
          <Image
            src={Fred}
            alt="The most beautiful dog!"
            width={300}
            height={500}
            className="about__img"
          />
          <p className="about__p">
            I am brazilian, born in Brazilia - DF. I&apos;ve lived for two
            months in Itapema/SC and almost two years in Andradas/MG.
          </p>
          <p className="about__p">
            I am {age()} years old, I like to watch movies and TV series, learn
            new things, read, dye my hair, a little of everything.
          </p>
          <p className="about__p">
            I&apos;m the mother of this beautiful dog called Fred Weasley.
          </p>
        </section>

        <section className="about__skills-sect">
          <h2 className="about__skills-title">My skills</h2>
          <ul className="about__skills-list">
            <li>Linux</li>
            <li>Git & GitHub</li>
            <li>HTML</li>
            <li>CSS</li>
            <li>JavaScript</li>
            <li>Jest</li>
            <li>React</li>
            <li>Next.js</li>
            <li>RTL</li>
            <li>SQL</li>
            <li>MongoDB</li>
            <li>Express</li>
            <li>REST API</li>
            <li>Sequelize</li>
            <li>Node.js</li>
            <li>Mocha</li>
            <li>Chai</li>
            <li>Python</li>
            <li>OOP</li>
            <li>SOLID</li>
          </ul>
        </section>
      </div>
    </S.Main>
  );
};

export default About;
