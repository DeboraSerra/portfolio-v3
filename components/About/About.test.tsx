import { render } from "@testing-library/react";
import About from "./About";
import TProvider from "../helpers/ThemeProvider";

describe("Test the About component", () => {
  it('Should match the snapshot', () => {
    const { asFragment } = render(<TProvider><About /></TProvider>)
    expect(asFragment()).toMatchSnapshot();
  })
})