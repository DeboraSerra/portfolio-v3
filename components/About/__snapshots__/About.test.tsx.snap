// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the About component Should match the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  min-height: 72vh;
  background-color: #DAD9DE;
  padding: 24px 48px;
  color: #47301E;
}

.c0 .about__title {
  margin-bottom: 48px;
  text-align: center;
  font-size: 40px;
  position: relative;
}

.c0 .about__title::before {
  content: "";
  background-color: #4E4946;
  position: absolute;
  bottom: -12px;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  width: 150px;
  height: 1px;
}

.c0 .about__left-sect {
  line-height: 1.6;
  text-align: center;
  overflow: hidden;
}

.c0 .about__p {
  margin-bottom: 12px;
  font-size: 16px;
  text-align: left;
}

.c0 .about__p::first-letter {
  font-size: 22px;
}

.c0 .about__img {
  float: right;
  height: 350px;
  object-fit: cover;
  object-position: top left;
  margin: 0 24px;
}

.c0 .about__skills-title {
  margin: 48px auto;
  position: relative;
  font-size: 26px;
  text-align: center;
}

.c0 .about__skills-title::before {
  content: "";
  background-color: #4E4946;
  position: absolute;
  bottom: -12px;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  width: 150px;
  height: 1px;
}

.c0 .about__skills-list {
  list-style-position: inside;
  list-style-type: circle;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  gap: 12px;
  max-height: 50vh;
}

.c0 .about__skills-list li {
  max-width: 200px;
  line-height: 2;
  font-size: 16px;
}

@media only screen and (max-width:576px) {
  .c0 {
    padding: 12px 24px;
  }

  .c0 .about__img {
    float: none;
    margin: 12px 0;
  }

  .c0 .about__skills-list {
    max-height: 80vh;
  }
}

<div
    class="c0"
  >
    <div
      class="container"
    >
      <h2
        class="about__title"
      >
        About me
      </h2>
      <section
        class="about__left-sect"
      >
        <img
          alt="The most beautiful dog!"
          class="about__img"
          data-nimg="1"
          decoding="async"
          height="500"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
          style="color: transparent;"
          width="300"
        />
        <p
          class="about__p"
        >
          I am brazilian, born in Brazilia - DF. I've lived for two months in Itapema/SC and almost two years in Andradas/MG.
        </p>
        <p
          class="about__p"
        >
          I am 30 years old, I like to watch movies and TV series, learn new things, read, dye my hair, a little of everything.
        </p>
        <p
          class="about__p"
        >
          I'm the mother of this beautiful dog called Fred Weasley.
        </p>
      </section>
      <section
        class="about__skills-sect"
      >
        <h2
          class="about__skills-title"
        >
          My skills
        </h2>
        <ul
          class="about__skills-list"
        >
          <li>
            Linux
          </li>
          <li>
            Git & GitHub
          </li>
          <li>
            HTML
          </li>
          <li>
            CSS
          </li>
          <li>
            JavaScript
          </li>
          <li>
            Jest
          </li>
          <li>
            React
          </li>
          <li>
            Next.js
          </li>
          <li>
            RTL
          </li>
          <li>
            SQL
          </li>
          <li>
            MongoDB
          </li>
          <li>
            Express
          </li>
          <li>
            REST API
          </li>
          <li>
            Sequelize
          </li>
          <li>
            Node.js
          </li>
          <li>
            Mocha
          </li>
          <li>
            Chai
          </li>
          <li>
            Python
          </li>
          <li>
            OOP
          </li>
          <li>
            SOLID
          </li>
        </ul>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
